r/R/metadata.R

```diff
@@ -175,7 +175,7 @@ apply_arrow_r_metadata <- function(x, r_metadata) {
       columns_metadata <- r_metadata$columns
       if (is.data.frame(x)) {
         # if columns metadata exists, apply it here
-        if (length(names(x)) && !is.null(columns_metadata)) {
+        if (length(names(x)) && !is.null(columns_metadata) && !all(map_lgl(columns_metadata, is.null))) {
```

Comment:
I know we don't want to benchmark this in our code, but this looks like something we want a test for, yeah?

Reply:
Oops, didn't think it was testable, but it totally is; added!